### ISSUE TITLE
Grails applications using this plugin and built via 'gradlew war' are not including all required resources

### ISSUE THREAD

**Issue by kdpeders** [grails/gorm-graphql#17]
In my application that uses this plugin I am packaging it via `Gradle`.
Packaged via `gradlew war` results in these jars getting added to `WEB-INF/lib`:

    -rw-r--r--    1 user1  group1   374K Mar 12 10:31 gorm-graphql-1.0.2.jar
    -rw-r--r--    1 user1  group1   647K Dec 20 10:26 graphql-java-6.0.jar

When deploying the Gradle built warfile I get the following error:

    ERROR g.b.c.GrailsApplicationPostProcessor Error loading spring/resources.groovy file: org/grails/gorm/graphql/plugin/GraphQLPostProcessor
    java.lang.NoClassDefFoundError: org/grails/gorm/graphql/plugin/GraphQLPostProcessor

Packaging via `grails war` results in these jars getting added to `WEB-INF/lib`:

    -rw-r--r--    1 user1  group1   557K Mar 12 10:31 gorm-graphql-1.0.2.jar
    -rw-r--r--    1 user1  group1   647K Dec 20 10:26 graphql-java-6.0.jar
    -rw-r--r--    1 user1  group1   374K Mar  8 19:09 org.grails-gorm-graphql-1.0.2.jar
    -rw-r--r--    1 user1  group1   557K Mar  8 19:09 org.grails.plugins-gorm-graphql-1.0.2.jar

The missing resource is in the `gorm-graphql-1.0.2.jar` file If and Only If it's build via `grails`.
I believe there is an incorrect dependency in the plugin that is preventing is from being built via 'gradlew'.

**Comment by kdpeders** [comment]
Note:  

https://repo.grails.org/grails/core/org/grails/gorm-graphql/1.0.2/ contains
`gorm-graphql-1.0.2.jar           08-Mar-2018 19:12  **374.33** KB`

https://repo.grails.org/grails/core/org/grails/plugins/gorm-graphql/1.0.2/ contains
`gorm-graphql-1.0.2.jar           08-Mar-2018 19:12  **556.52** KB`

I need to generate my war file via `gradle war` so that it's not runnable (as I deploy to a tomcat container and the runnable war files don't deploy correctly)

**Comment by orr3** [comment]
@kdpeders 
Hi! Were you able to resolve this? 
Running locally all gorm-graphql plugins are included and I can query the endpoints. But running `grails war` produces a package without these plugins.

**Comment by kirpi4ik** [comment]
@orr3  first you could try to change the copy strategy 
war {
    duplicatesStrategy = DuplicatesStrategy.FAIL
}
My assumption that it will fail because the gorm plugin has same `artifactId` with the core library, so even the `groupId` is different in the end gradle will use only the `artifactId` to archive the fat jar/war/bootJar. With duplicated jar inside uberjar(which is not forbidden) the behaviour would be very unpredictable.

**Comment by jameskleeh** [comment]
I'm open for any suggestions on how to fix this

**Comment by kirpi4ik** [comment]
@jameskleeh Although it is not the best way but I assume that one option would be to use unique name [here](https://github.com/grails/gorm-graphql/blob/master/gradle/common-publishing.gradle#L90) .
In the meantime as workaround @orr3 you can rename the final artifact name by including also the groupId . E.g. for `bootJar` it would be like :
```
bootJar {
    rootSpec.filesMatching('**/*.jar', { jar ->
        String groupId = jar.file.parentFile.parentFile.parentFile.parentFile.name
        jar.name = "$groupId-${jar.name}"
    })
}
```

**Comment by orr3** [comment]
@kirpi4ik thanks for the suggestions. I will look into this. Since we use this plugin in a new project we ended up to downgrading from Grails `4.0.1` to `3.3.10` which resolved the problem.

I don't know if another gradle version is used in `3.3.10`, but surely `grails war` don't have a problem packaging resources with the same artifactId.

**Comment by ianibo** [comment]
Just run into this with grails 4.0.1 also - it appears that the plugin org.grails.plugins:gorm-graphql:1.0.3.BUILD-SNAPSHOT includes org.grails:gorm-graphql:1.0.3.BUILD-SNAPSHOT as a dependency but when these get bundled up they both get included as only the jar name -- "gorm-graphql:1.0.3.BUILD-SNAPSHOT.jar" - ending up with a libs like this:

570018 Sun Dec 29 14:54:00 GMT 2019 WEB-INF/lib/gorm-graphql-1.0.3.BUILD-SNAPSHOT.jar
383317 Sun Dec 29 14:54:00 GMT 2019 WEB-INF/lib/gorm-graphql-1.0.3.BUILD-SNAPSHOT.jar
1297269 Sun Dec 29 14:54:00 GMT 2019 WEB-INF/lib/graphql-java-13.0.jar

I'm guessing that later on only one of these jar files wins and either way we end up with missing classes. I wonder if it might be wise to give the two artifacts different names, rather than relying upion the group to differentiate them?

**Comment by kirpi4ik** [comment]
@ianibo although it look like the easiest way in this particular case would be to use unique `artifactId` that is not really a solution for your project. There is no conventions that artifactIds must be unique in the maven/gradle world so there will be always a chance that you will have in your project identical names, where `groupId` has exactly the purpose to help defining a unique artifact. 
So even if having unique `artifactId` is a "nice to have" I would say this is more an issue of the gradle task which generates the fat jar by setting inner jar names by only using artifactId.
